src/infrastructure/ServiceBase.ts

```diff
@@ -0,0 +1,45 @@
+import { v4 as uuid } from 'uuid';
+import { CrudService } from "./CrudService";
+import moment, { Moment, parseZone } from 'moment';
+import { setTime } from '../common/TimeUtils'
+import { AutoWired, Inject } from 'typescript-ioc';
+import { CurrentUser } from './CurrentUser';
+
+@AutoWired
+export abstract class ServiceBase<T> extends CrudService<T> {
+    @Inject
+    private _currentUser!:CurrentUser;
+    protected currentUser(){
+        return this._currentUser;
+    }
+
+    getAll(): Promise<T[]> {
+        throw new Error('Method not implemented.');
```

Comment:
No concerns with these not being implemented?

Reply:
Nope, all of our services generally inherit from a service that has implemented these methods (i.e DatabaseService).  These used to be abstract, but then I wanted to add some default injected members (i.e. currentUser) to the base class.  We write tests for all of our api endpoints so these errors would only come up if someone extends ServiceBase directly (which is only happening in one place `UserService` and it doesn't need the base methods).  Good question though.

Reply:
👍